hypothesis-python/src/hypothesis/core.py

```diff
@@ -424,7 +427,7 @@ def __init__(self, filename, source, target):
         self.target = target
 
 
-ARC_CACHE = {}
+ARC_CACHE = {}  # type: Dict[str, Dict[Any, Dict[Any, Arc]]]
```

Comment:
It seems like we're ending up on type hints for a lot of things that are never exposed in the public API. Why is that?

Reply:
Bizarre design choice - Mypy will error on any global collection because it can't infer the element type, and there's no way to disable this or make it infer Generic[Any]